### PR TITLE
Adds Javascript API to customize the font picker

### PR DESCRIPTION
**Pull request by radazzouz**
Solves #320 

---

# Details

Adds Javascript API to customize the font picker.

The exposed React Native props:

```js
  /**
   * availableFontNames: Can be used to specfiy the available font names in the font picker.
   * Note: You need to set the desired font family names as `UIFontDescriptor`. See https://developer.apple.com/documentation/uikit/uifontdescriptor?language=objc
   * See `CustomFontPicker` in https://github.com/PSPDFKit/react-native/blob/master/samples/Catalog/Catalog.ios.js
   *
   * @platform ios
   */
  availableFontNames: PropTypes.array,
  /**
   * selectedFontName: Can be used to specfiy the current selected font in the font picker.
   * Note: You need to set the desired font family name as `UIFontDescriptor`. See https://developer.apple.com/documentation/uikit/uifontdescriptor?language=objc
   * See `CustomFontPicker` in https://github.com/PSPDFKit/react-native/blob/master/samples/Catalog/Catalog.ios.js
   *
   * @platform ios
   */
  selectedFontName: PropTypes.string,
  /**
   * showDownloadableFonts: Can be used to show or hide the downloadable fonts section in the font picker. Defaults to showing a the downloadable fonts (true).
   * See `CustomFontPicker` in https://github.com/PSPDFKit/react-native/blob/master/samples/Catalog/Catalog.ios.js
   *
   * @platform ios
   */
  showDownloadableFonts: PropTypes.bool,
```

recording

# Acceptance Criteria

- [x] Add Javascript API to customize the font picker on Android.
- [x] Add Javascript API to customize the font picker on iOS.
- [ ] When approved, right before merging, rebase with master and increment the package version in `package.json`, `package-lock.json`, `samples/Catalog/package.json`, and `samples/NativeCatalog/package.json` (see example commit:  https://github.com/PSPDFKit/react-native/pull/202/commits/1bf805feef2ac268743e4905d94d6d8c8f16ec59).
- [ ] Create a new release (and tag) with the new package version (see https://github.com/PSPDFKit/react-native/releases).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -661,18 +661,19 @@ PSPDFKitView.propTypes = {
...
   availableFontNames: PropTypes.array,
   /**
    * selectedFontName: Can be used to specfiy the current selected font in the font picker.
-   * Note: You need to set the desired font family name as `UIFontDescriptor`. See https://developer.apple.com/documentation/uikit/uifontdescriptor?language=objc
+   *
+   * Note on iOS: You need to set the desired font family name as `UIFontDescriptor`. See https://developer.apple.com/documentation/uikit/uifontdescriptor?language=objc
    * See `CustomFontPicker` in https://github.com/PSPDFKit/react-native/blob/master/samples/Catalog/Catalog.ios.js
    *
-   * @platform ios
+   * Note on Android: This is the default font that is selected, if the users changes the font that will become the new default.
```
Comment (radazzouz) [inline index.js:676]: We have the same behaviour on iOS. Thanks for adding this.

```suggestion
   * Note: This is the default font that is selected, if the users changes the font that will become the new default.
```

Comment (irgendeinich) [reply index.js:676]: In that case we could consider renaming it to `defaultFontName` instead of `selectedFontName`. I think that would be much clearer.

Comment (radazzouz) [reply index.js:676]: Sorry I was wrong, on iOS `selectedFontName` only sets the selected font name in the picker. It does not change the default font for free text annotations.

File: ios/RCTPSPDFKit/RCTPSPDFKitViewManager.m [hunk 2]
```diff
@@ -107,6 +121,42 @@ @implementation RCTPSPDFKitViewManager
...
 
+RCT_CUSTOM_VIEW_PROPERTY(availableFontNames, NSArray, RCTPSPDFKitView) {
+  if (json && [RCTConvert NSArray:json]) {
+    view.availableFontNames = [RCTConvert NSArray:json];
+    staticAvailableFontNames = view.availableFontNames;
+
+    // We use a subclassed font picker view controller to customize it.
+    [view.pdfController updateConfigurationWithoutReloadingWithBuilder:^(PSPDFConfigurationBuilder * _Nonnull builder) {
+      [builder overrideClass:PSPDFFontPickerViewController.class withClass:CustomFontPickerViewController.class];
+    }];
```
Comment (steviki) [inline ios/RCTPSPDFKit/RCTPSPDFKitViewManager.m:132]: Could we always use the custom override, so we don't need to update the configuration all the time one of these properties is set, and just fallback to the default behavior, when none of these variables are set?

Comment (radazzouz) [reply ios/RCTPSPDFKit/RCTPSPDFKitViewManager.m]: Good point! 👍 Addressed in https://github.com/PSPDFKit/react-native/pull/363/commits/cc92d1ad000c4af1b4951752b84e4f5e0734c4bd

**Pull request merged**